source/PreferencesPanel.cpp

```diff
@@ -179,6 +205,31 @@ bool PreferencesPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &comma
 		if(zones[latest].Value().KeyName() != Command::MENU.KeyName())
 			Command::SetKey(zones[latest].Value(), 0);
 	}
+	else if(key == 'd')
+	{
+		page = 'i';
+		if(!downloadedInfo)
+		{
+			if(!Plugins::Download(PLUGIN_LIST_URL, Files::Config() + "plugins.json"))
+				printf("Failed to download Plugin-List from %s\n", PLUGIN_LIST_URL.c_str());
+			ifstream pluginlistFile(Files::Config() + "plugins.json");
+			pluginInstallList = nlohmann::json::parse(pluginlistFile);
```

Comment:
Downloading + Parsing might take a while and doing it here will freeze the game for a bit, so I would say that this should also be done async. And while that happens the player should also see a message, similar to above, that the file is being downloaded.

Reply:
ok